concent_api/core/views.py

```diff
@@ -426,6 +232,18 @@ def handle_send_ack_report_computed_task(request, client_message):
             requestor_public_key = client_public_key,
             status               = ReceiveStatus,
         )
+        subtask = update_subtask(
+            subtask_id                  = client_message.task_to_compute.compute_task_def['subtask_id'],
+            requestor_public_key        = client_public_key,
+            state                       = Subtask.SubtaskState.REPORTED,
+            ack_report_computed_task    = client_message,
+        )
+        store_pending_message(
+            response_type       = PendingResponse.ResponseType.AckReportComputedTask,
+            client_public_key   = subtask.provider.public_key_bytes,
```

Comment:
The ack should be sent to the provider inside `ForceReportComputedTaskResponse`.

Reply:
We currently don't handle this kind of response, so I guess you are referring to `RejectReportComputedTask` ?

:+1: I wasn't aware of this, we should create separate issue for creating test for this scenario.

Reply:
I thought this was already done in #87? This task was all about `ForceReportComputedTaskResponse` and it's marked as done.

Reply:
But we support this kind of message. If `Ack` is message in queue then:
```python
elif database_message.response_type == PendingResponse.ResponseType.AckReportComputedTask.name:  # pylint: disable=no-member
        if database_message.subtask.ack_report_computed_task is not None:
            ack_report_computed_task = deserialize_message(database_message.subtask.ack_report_computed_task.data.tobytes())
            response_to_client = message.concents.ForceReportComputedTaskResponse(
                ack_report_computed_task = ack_report_computed_task,
            )
            mark_message_as_delivered_and_log(database_message, response_to_client)
            return response_to_client
```

Reply:
OK. So the response type in queue is wrong. It should always match the message that will be returned to the client.